circuits/cpp/barretenberg/cpp/src/barretenberg/honk/proof_system/eccvm_prover.cpp

```diff
@@ -0,0 +1,385 @@
+#include "eccvm_prover.hpp"
+#include "barretenberg/honk/pcs/claim.hpp"
+#include "barretenberg/honk/pcs/commitment_key.hpp"
+#include "barretenberg/honk/proof_system/lookup_library.hpp"
+#include "barretenberg/honk/proof_system/permutation_library.hpp"
+#include "barretenberg/honk/proof_system/prover_library.hpp"
+#include "barretenberg/honk/sumcheck/sumcheck.hpp"
+#include "barretenberg/honk/utils/power_polynomial.hpp"
+#include "barretenberg/polynomials/polynomial.hpp"
+#include "barretenberg/polynomials/univariate.hpp" // will go away
+#include "barretenberg/proof_system/relations/lookup_relation.hpp"
+#include "barretenberg/proof_system/relations/permutation_relation.hpp"
+#include "barretenberg/transcript/transcript_wrappers.hpp"
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <span>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace proof_system::honk {
+
+/**
+ * Create ECCVMProver_ from proving key, witness and manifest.
+ *
+ * @param input_key Proving key.
+ * @param input_manifest Input manifest
+ *
+ * @tparam settings Settings class.
+ * */
+template <ECCVMFlavor Flavor>
+ECCVMProver_<Flavor>::ECCVMProver_(std::shared_ptr<typename Flavor::ProvingKey> input_key,
+                                   std::shared_ptr<PCSCommitmentKey> commitment_key)
+    : key(input_key)
+    , queue(commitment_key, transcript)
+    , pcs_commitment_key(commitment_key)
+{
+
+    // TODO(@zac-williamson) Future work; is there a cleaner way of doing this?
```

Comment:
Could you make an issue?

Reply:
✅ [#2213](https://github.com/AztecProtocol/aztec-packages/issues/2213)